satpy/enhancements/__init__.py

```diff
@@ -124,6 +124,9 @@ def func(band_data, index=None):
                                  attrs=band_data.attrs)
         return band_data
 
+    # mask reflectances that are below 0
+    bad_mask = (img.data < 0).any(axis=0)
```

Comment:
Won't that create holes in composite in very dark places ?

Reply:
You are absolutely right. This turned in to a WIP after thinking more about it and testing things. That's why I asked the question about this on slack. The cira stretch and the crefl stretch behave differently regarding negative reflectances (cira does log on them so they become NaN). I am starting to think that the clipping should just be handled like in the above line. That is, if any reflectance is negative, clip all of them to 0 (black). Otherwise, this clipping could be made part of the rayleigh correction or the ratio sharpening code. If the bands don't share "special" pixels together then there are some weird effects on the terminator limb (like bright cyan pixels).

Bottom line: At night the dark parts of the earth are showing up as negative reflectances, should they be black or transparent. A negative reflectance isn't really a real reflectance, right?